MQTT Bridge Config via UI

dj3mu opened the pull request:
### Implementation of the discussions of [this](https://openwb.de/forum/viewtopic.php?f=6&t=591) openWB forum thread.

#### Introducing a settings GUI for configuration of MQTT bridges (openWB Mosquitto as client for the remote MQTT server).

Bridge configuration for UI display is directly read from `99-bridge-*` files in `/etc/mosquitto/conf.d`.

When form is submitted, the configuration is first stored in openWB's `ramdisk` folder. From there it gets picked up by a script called with every `regel.sh` run. This is required for the PHP scripts not having sufficient access rights to modify `/etc/mosquitto/conf.d`.

Deletion of bridges is performed by writing a list of configs to be deleted into a file called `ramdisk/99-bridgesToDelete` (one per line). Actual deletions are then executed as part of `regel.sh` run, too.

**Note:** The transmitted debug file has been extended by the bridge configurations with passwords and comments stripped off.

There's a lot of space for improvements and I'm open for all suggestions.  
Also testing by other users would be very welcome.

Feedback can be provided via Github or openWB Forum.

snaptec commented:
Looks good, thanks in advance!
I would really love not to use the regel.sh interval for checking.
Best way:
from php trigger a "runs/checkmqttconf.sh"
from there you can run everything.

You may have a look at:
web/tools/smashm.php
and the .sh
runs/smashm.sh

dj3mu commented:
Should now all be done as requested in [previous comment](https://github.com/snaptec/openWB/pull/267#issuecomment-560523931).

Thanks for the hints how to do it. That helped and really saved time for reverse-engineering.

snaptec commented:
A Bug:
Remove the check enabled on "Brücke" and save does not remove the activation. As a result a .conf and .no are in /etc/mosquitto/conf.d
Also deactivating one bridge results in 2 bridges, one activated, one deactivated.
I would prefer as a first step to just allow only one bridge configuration. 
It's to confusing at the moment I think.

MQTT config itself is fine, used the same here :)

dj3mu commented:
> A Bug:
> Remove the check enabled on "Brücke" and save does not remove the activation. As a result a .conf and .no are in /etc/mosquitto/conf.d
> Also deactivating one bridge results in 2 bridges, one activated, one deactivated.

Oops. I thought that I had tried that.  
Will investigate again and report back.

> I would prefer as a first step to just allow only one bridge configuration.
> It's to confusing at the moment I think.

Would it be ok for you if multiple `99-bridge-*` files are already present in the `conf.d` folder (no matter how they made it in there) all of them will get displayed?  
In other words: Is it ok if I only lock down the creation of >1 config. But if more configs are already present (e.g. by manual user change) they will still be shown?
Simple reason: That would be easier to implement and less ambiguous. For multiple configs it else wouldn't be clear which one to show.

snaptec commented:
> Would it be ok for you if multiple 99-bridge-* files are already present in the conf.d folder (no matter how they made it in there) all of them will get displayed?

Thats ok if it is not possible, like through the bug, to create more than one in the ui.

snaptec commented:
..changing name of existing conf results in one more...

dj3mu commented:
Implemented allowing a single config only. If there's any config found already the section to add a new config will not be shown anymore.

But sorry, I cannot reproduce the issue with duplicating the config my openWB. And I've verified again that I'm having exactly that version on this branch.  
Actually the code is now such that the files matching the saved config will always get deleted and re-written. So do the files of old names in case the config gets renamed.

To track the issue, I've now added redirecting output of the `runs/checkmqttconf.sh` into a file `ramdisk/checkmqttconf.log`.

Can you please re-test using "Debug >= 1" and the latest commit b5e4b2edee80f664c7085bc10c080833ebbb5d42 and share the file `ramdisk/checkmqttconf.log`, the `99-bridge-*` files in `/etc/mosquitto/conf.d` (strip off passwords)?
